core/src/test/java/com/amplifyframework/core/AmplifyTest.java

```diff
@@ -90,8 +91,32 @@ public void reconfigurationAttemptRaisesException() throws AmplifyException {
             actuallyThrown.getMessage()
         );
         assertEquals(
-            "Remove the duplicate call to `Amplify.configure()`",
+            "Remove the duplicate call to `Amplify.configure()`.",
             actuallyThrown.getRecoverySuggestion()
         );
     }
+
+    /**
+     * It is an error to call {@link Amplify#addPlugin(Plugin)}} after configuration.
+     * @throws Amplify.AlreadyConfiguredException
+     *  NOTE: The name of this method must result in it running last, according to {@link FixMethodOrder}
+     */
+    @Test
+    public void secondaryAddPluginAfterConfigurationRaisesException() throws Amplify.AlreadyConfiguredException {
```

Comment:
Hmm, I don't see any calls to `Amplify.configure()` in this test.  Don't you need to do that before calling `Amplify.addPlugin(...)` since that is the scenario being tested?

Reply:
I believe that the configuration status is retained from the previous test, `reconfigurationAttemptRaisesException()`.

Reply:
Ahh, that makes sense.  I think this is an unfortunate side effect of the fact that the `Amplify` facade is just not designed in a very testable manner.  

Is there a way we could implement an `@After` method which resets the configuration?   I think it would need to use reflection to set the [static `CONFIGURATION_LOCK` property](https://github.com/aws-amplify/amplify-android/blob/682c9e778adac74c2261bce4de4866c5e255a5eb/core/src/main/java/com/amplifyframework/core/Amplify.java#L83) of `Amplify` back to false.  

Something like 
```
@After
public void cleanup() {
    Field configurationLockField = Field.class.getField("CONFIGURATION_LOCK");

    // the field is `private`, so make it accessible
    configurationLockField.setAccessible(true);

    // remove the `final` modifier so we can change the value.
    Field modifiersField = Field.class.getDeclaredField("modifiers"); 
    modifiersField.setAccessible(true);
    modifiersField.setInt(configurationLockField, configurationLockField.getModifiers() & ~Modifier.FINAL);

   // Now actually reset the value to false.
   configurationLockField.set(null, new AtomicBoolean(false));
}
```

If all that ends up being too complicated or not working for some reason, I think we could merge this as is, but it would be good to at least attempt a better approach here.

Reply:
Chatted offline and determined we weren't able to get this reflection approach to work, so I've approved this PR.